### PR TITLE
Update ldms yaml updater mode definition

### PR DESCRIPTION
**Pull request by nick-enoent**
mode options are : pull, auto, push, onchange
pull - default, updater will schedule set updates according to defined
interval and offset values
auto - updater will schedule set updates according to update hint
push - updater will receive an update only when the set source
explicitly pushes an update
onchange - updater will get an update whenever the set source ends a
transaction or pushes an update

### CODE REVIEW

**Comment by tom95858** [comment]
@nichamon could you please review this?

**Comment by narategithub** [comment]
@nick-enoent this does not break the existing basic maestro and maestro+raft tests.

File: src/maestro/Communicator.py [hunk 1]
```diff
@@ -695,18 +695,16 @@ def updtr_add(self, name, interval=None, offset=None, push=None, auto=None, perm
                 LDMSD_Req_Attr(attr_id=LDMSD_Req_Attr.INTERVAL, value=str(interval)),
```
Comment (nichamon) [inline src/maestro/Communicator.py:695]: I want to point out that 'updtr_add' requires the 'interval' attribute. You may want to consider changing the signature of the function. It is orthogonal to the patch, though

Comment (nick-enoent) [reply src/maestro/Communicator.py:695]: Thanks for looking at this @nichamon. I'll update the Communicator method to require the interval parameter, and add an example in the config/new_config.yaml template

Comment (nichamon) [reply src/maestro/Communicator.py:695]: Let's remove the ```if interval``` condition.

**Review by nichamon — APPROVED** [review]

**Pull request merged**